crates/runtime/src/datafusion/query.rs

```diff
@@ -116,9 +124,63 @@ impl Query {
     /// # Panics
     ///
     /// Panics when running under test if no cache key is computed for the query.
-    #[allow(clippy::too_many_lines)]
     pub async fn run(self) -> Result<QueryResult> {
         let request_context = RequestContext::current(AsyncMarker::new().await);
+        let runtime_handle = self.df.tokio_runtime().clone();
+
+        let (batch_tx, batch_rx) = mpsc::channel::<Result<RecordBatch, DataFusionError>>(2);
+        let (result_tx, result_rx) = oneshot::channel::<Result<(CacheStatus, SchemaRef), Error>>();
+
+        let request_context_clone = Arc::clone(&request_context);
+        let driver_task = async move {
+            match self.run_internal(request_context_clone).await {
+                Ok(query_result) => {
+                    let schema = query_result.data.schema();
+                    let cache_status = query_result.cache_status;
+
+                    if result_tx.send(Ok((cache_status, schema))).is_err() {
+                        return;
+                    }
+
+                    let mut stream = query_result.data;
+                    while let Some(batch) = stream.next().await {
+                        if batch_tx.send(batch).await.is_err() {
+                            break;
+                        }
+                    }
+                }
+                Err(err) => {
+                    if result_tx.send(Err(err)).is_err() {
+                        tracing::debug!("Failed to send query result; receiver dropped");
+                    }
+                }
+            }
+        };
+
+        let driver_handle = runtime_handle.spawn(driver_task);
```

Comment:
Will this not make us lose current `request_context` as it will run on a different thread pool? Shouldn't we wrap executing this task with current `request_context`'s scope?